lib/skate_web/channels/vehicles_channel.ex

```diff
@@ -1,8 +1,12 @@
 defmodule SkateWeb.VehiclesChannel do
   use SkateWeb, :channel
 
+  alias Phoenix.Socket
+  alias Realtime.{Server, Vehicle}
+
+  @spec join(topic :: String.t(), payload :: map, socket :: Socket.t()) ::
```

Comment:
These already have types: https://github.com/phoenixframework/phoenix/blob/v1.4.3/lib/phoenix/channel.ex#L288

So you can simplify to `@impl Phoenix.Channel`

Reply:
Great, thanks, I didn't know that.

Reply:
Anything that has a behaviour works this way: GenServer, GenStage, and Plug are some other examples off the top of my head.